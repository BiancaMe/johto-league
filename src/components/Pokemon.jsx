import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

export const filterType = (poke, id) => {
  let bool = false;
  if (poke) {
    const res = poke.types.map((aux) => {
      if (aux.type.name === id) { return true; }
      return false;
    });

    for (let i = 0; i <= res.length; i += 1) {
      if (res[i] === true) {
        bool = true;
        break;
      }
    }
  }
  return bool;
};

const Pokemon = ({ id }) => {
  const pokemon = useSelector(
    (state) => state.pokemon.pokemon,
  );

  return (
    <>
      <div className="clasification">
        <p className="c-text">
          {' '}
          POKEMON
          {' '}
          {id}
        </p>
      </div>
      <div className="pokemon-grid">
        {pokemon.map((poke) => {
          const res = filterType(poke, id);
          if (res === true) {
            return (
              <div className="pokemon" key={poke.name}>
                <img
                  src={poke.sprites.other.dream_world.front_default}
                  alt={poke.name}
                  className="poke-img"
                />
                <div className="poke-info">
                  <h2 className="p-name">{poke.name}</h2>
                  <p className="p-data">
                    -Weight :
                    {' '}
                    {poke.weight}
                  </p>
                  <p className="p-data">
                    -Ability :
                    {' '}
                    {poke.abilities.map((aux) => aux.ability.name)[0]}
                  </p>
                </div>
              </div>
            );
          }
          return ('');
        })}
      </div>
    </>
  );
};

Pokemon.propTypes = {
  id: PropTypes.string.isRequired,
};

export default Pokemon;
