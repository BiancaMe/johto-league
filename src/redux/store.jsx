import { configureStore } from '@reduxjs/toolkit';
import pokemonSlice from './pokemon/pokemonSlice';

const store = configureStore({
  reducer: {
    pokemon: pokemonSlice,
  },
});

export default store;
