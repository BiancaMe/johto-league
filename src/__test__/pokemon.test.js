import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Pokemon from '../components/Pokemon';
import store from '../redux/store';

describe('Functions in Pokemon Component', () => {
  it('Get pokemon', async () => {
    const poke = renderer.create(
      <Provider store={store}>
        <Pokemon id="normal" />
      </Provider>,
    );

    expect(poke).toMatchSnapshot();
  });
});
