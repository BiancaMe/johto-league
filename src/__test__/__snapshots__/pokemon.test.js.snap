// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functions in Pokemon Component Get pokemon 1`] = `
Array [
  <div
    className="clasification"
  >
    <p
      className="c-text"
    >
       
      POKEMON
       
      normal
    </p>
  </div>,
  <div
    className="pokemon-grid"
  />,
]
`;

exports[`Pokemon render`] = `
<div className="clasification">
<p className="c-text"> POKEMON normal</p></div>,
 <div className="pokemon-grid" />
`;
